traitcuration/traits/datasources/zooma.py

```diff
@@ -0,0 +1,84 @@
+"""
+This module contains all the functionality that uses ZOOMA to retrieve mapping suggestions and create the suggested
+terms in the app's database
+"""
+import requests
+
+from ..models import Trait, MappingSuggestion, OntologyTerm, User
+from .ols import make_ols_query, get_ontology_id
+
+
+BASE_URL = "http://www.ebi.ac.uk/spot/zooma/v2/api"
+
+
+def get_zooma_suggestions():
+    """
+    Retrieves zooma mapping suggestions for all traits and creates terms and suggestions in the app's database for each
+    of them.
+    """
+    traits = Trait.objects.all()
+    for trait in traits:
+        results_list = get_suggestions_for_trait(trait)
+        for result in results_list:
+            term = create_local_term(result)
+            # If local term creation returns None, it means that this term already exists in the database
+            if term is not None:
+                create_mapping_suggestion(trait, term)
+
+
+def get_suggestions_for_trait(trait):
+    """
+    Takes in a trait object as its argument and returns the zooma response of a suggestion list as a dictionary.
+    """
+    formatted_trait_name = trait.name.replace(' ', '+')
+    response = requests.get(
+        f"{BASE_URL}/services/annotate?propertyValue={formatted_trait_name} \
+        &filter=required:[none],ontologies:[efo,mondo,hp,orphanet]")
+    return response.json()
+
+
+def create_local_term(result):
+    """
+    Takes in a single zooma suggestion result and creates an ontology term for it in the app's database.
+    """
+    term_iri = result["semanticTags"][0]
+    print(term_iri)
+    if OntologyTerm.objects.filter(iri=term_iri).exists():
+        return
+    # Search for the term in EFO and return its information. If it is not in EFO, search for it in original ontology.
+    term_info = make_ols_query(term_iri, 'efo')
+    term_ontology_id = 'efo'
+    # None here means that the term wasn't found in the EFO ontology
+    if term_info is None:
+        term_ontology_id = get_ontology_id(term_iri)
+        term_info = make_ols_query(term_iri, term_ontology_id)
+        # If the term is not found in the original ontology either, return
```

Comment:
This situation (which is really unusual) should immediately change the term status to "deleted" (we do have "deleted" in the final db model, right?)

Reply:
Yeah we do, but do we still want to create the term if we don't find it anywhere during OLS queries?

Reply:
Yes, I would say we still want to create this term. This situation means that either something is wrong on the ZOOMA side, or some bug happened on our side during OLS lookup. So it's better to have this more explicit (by showing up as a deleted term in a suggestion), so this could be noticed and fixed/reported.